contents/code/bladelayout.js

```diff
@@ -52,20 +52,19 @@ BladeLayout.prototype.addTile = function() {
             return;
         } else {
             // Divide the screen width evenly between full-height tiles
-            var lastRect = this.tiles[0].rectangle;
-            var newRect = Qt.rect(this.screenRectangle.x,
-                                  lastRect.y,
-                                  (this.screenRectangle.width + this.screenRectangle.x) / (this.tiles.length + 1), 
+            var tileWidth = Math.floor(this.screenRectangle.width / (this.tiles.length + 1));
+            var newRect = Qt.rect(this.screenRectangle.x + this.tiles.length * tileWidth,
```

Comment:
Note: The old code inserted the new rect to the left, this inserts it to the right.

I don't think it's vitally important either way, and inserting to the right is more in line with what the other layouts do.